pkg/policyengine/connectivitypdp/connectivity_pdp.go

```diff
@@ -101,38 +101,28 @@ func (pdp *PDP) GetPolicies() []v1alpha1.AccessPolicy {
 	return res
 }
 
-// AddOrUpdatePolicy adds a PrivilegedAccessPolicy to the PDP.
+// AddOrUpdatePolicy adds an AccessPolicy or a PrivilegedAccessPolicy to the PDP.
 // If a policy with the same name already exists in the PDP,
 // it is updated (including updating the Action field).
 // Invalid policies return an error.
-func (pdp *PDP) AddOrUpdatePrivilegedPolicy(policy *v1alpha1.PrivilegedAccessPolicy) error {
+func (pdp *PDP) AddOrUpdatePolicy(policy *v1alpha1.AccessPolicy, privileged bool) error {
 	if err := policy.Spec.Validate(); err != nil {
 		return err
 	}
 
-	policyName := types.NamespacedName{Namespace: policy.Namespace, Name: policy.Name}
-	pdp.privilegedPolicies.addPolicy(policyName, &policy.Spec)
-	return nil
-}
-
-// AddOrUpdatePolicy adds an AccessPolicy to the PDP.
-// If a policy with the same name already exists in the PDP,
-// it is updated (including updating the Action field).
-// Invalid policies return an error.
-func (pdp *PDP) AddOrUpdatePolicy(policy *v1alpha1.AccessPolicy) error {
-	if err := policy.Spec.Validate(); err != nil {
-		return err
+	polName := types.NamespacedName{Namespace: policy.Namespace, Name: policy.Name}
+	if privileged {
+		pdp.privilegedPolicies.addPolicy(polName, &policy.Spec)
+	} else {
+		pdp.regularPolicies.addPolicy(polName, &policy.Spec)
 	}
-
-	policyName := types.NamespacedName{Namespace: policy.Namespace, Name: policy.Name}
-	pdp.regularPolicies.addPolicy(policyName, &policy.Spec)
 	return nil
 }
 
 // DeletePolicy deletes an AccessPolicy with the given name and privilege from the PDP.
 // If no such AccessPolicy exists in the PDP, an error is returned.
-func (pdp *PDP) DeletePolicy(policyName types.NamespacedName, privileged bool) error {
-	if privileged {
+func (pdp *PDP) DeletePolicy(policyName types.NamespacedName) error {
+	if policyName.Namespace == "" { // privileged policies have no namespace, regular policies do
```

Comment:
Please confirm that k8s indeed removes it since the metadata has a field for it regardless.

Reply:
Added back the `privileged` parameter to this function